pkg/controllers/managementuser/rbac/globalrolebinding_handler.go

```diff
@@ -258,3 +228,36 @@ func grbByUserAndRole(obj interface{}) ([]string, error) {
 
 	return []string{rbac.GetGRBTargetKey(grb) + "-" + grb.GlobalRoleName}, nil
 }
+
+// newCRTBtoGRBResolver returns a resolver which provides the key to the GRB that owns a given CRTB if one exist.
+func newCRTBtoGRBResolver(grbCache mgmtv3.GlobalRoleBindingCache) relatedresource.Resolver {
+	return func(_, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
+		crtb, ok := obj.(*v3.ClusterRoleTemplateBinding)
+		if !ok || crtb == nil {
+			return nil, nil
+		}
+
+		var grbOwner *v1.OwnerReference
+		for i := range crtb.OwnerReferences {
+			if crtb.OwnerReferences[i].Kind == "GlobalRoleBinding" {
```

Comment:
Please check the APIVersion here as well to make sure that it comes from the `management.cattle.io/v3` api group/version.

Reply:
done.